### ISSUE TITLE
rule(md044): allow configuration of rule for frontmatter

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#743]
### Problem or motivation

Rule MD044 (https://github.com/rvben/rumdl/blob/50035a41350da1b037486cfa1e9ade36dcb00801/docs/md044.md) allows `rumdl` to check for required capitalisation of names, both in the the body of Markdown files and in the frontmatter.

This is very helpful because prose can be in both the Markdown body and frontmatter.

However, sometimes frontmatter is used for content other than prose, which can lead to false positives.

### Proposed solution

It should be possible to exclude certain named frontmatter fields from rule MD044.  For example:

- Take this Markdown:

  ```markdown
  ---
  title: Heading for MyApp
  description: A description for MyApp
  link: 'this/is/a/link/to/myapp.md'
  ---

  This is a [link for `myapp`](this/is/a/link/to/myapp.md).
  ```

- Take this `rumdl` configuration:

  ```toml
  [global]
  enable = ["MD044"]

  [MD044]
  names = ["MyApp"]
  ```

At the moment, `rumdl` will find a violation on line 4. Ideally, `link` could be excluded from the check because, unlike `title` and `description`, there is no value in linting it.

I propose that a new configuration item (an array) be added to either:

- List frontmatter fields that should be checked. All other fields are ignored.
- List frontmatter fields that should be excluded. All other fields are checked.

### Alternatives considered

My proposal extends rule MD044. Some other possibilities:

1. A new rule that would be a central place to configure all front-matter checking for `rumdl`. Rule MD071 (https://github.com/rvben/rumdl/blob/50035a41350da1b037486cfa1e9ade36dcb00801/docs/md071.md) and MD072 (https://github.com/rvben/rumdl/blob/50035a41350da1b037486cfa1e9ade36dcb00801/docs/md072.md) are existing rules that are dedicated to frontmatter, but neither seem like they could be extended in the way described here.
1. New global options (perhaps called `include-frontmatter` and `exclude-frontmatter`) that would turn checking on and off for specified fields.

I'm not sure how likely other rules will cause false positives on frontmatter field values. I guess if it's more than one, then a new rule or global option is preferable. If this is likely just a problem with MD044, then extending it is preferable.

### Feature category

New lint rule

**Comment by rvben** [comment]
Shipped in v0.2.38 - MD044 can now be configured for frontmatter. Thanks @eread.
